server/services/meilisearch/config.js

```diff
@@ -94,12 +94,14 @@ module.exports = ({ strapi }) => {
           Array.isArray(entries) &&
           typeof contentTypeConfig?.filterEntry === 'function'
         ) {
-          const filtered = entries.filter(entry =>
-            contentTypeConfig.filterEntry({
+          const filtered = await entries.reduce(async (memo, entry) => {
+            const result = await contentTypeConfig.filterEntry({
               entry,
               contentType,
             })
-          )
+            if (!result) return memo
+            return [...(await memo), entry]
```

Comment:
I tried some ideas to don't keep this `...(await memo)` it seems weird, but I was not able to improve :/